tasks/check_crl_cert.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+declare PT__installdir
+# shellcheck disable=SC1090
+source "$PT__installdir/ca_extend/files/common.sh"
+PUPPET_BIN='/opt/puppetlabs/puppet/bin'
+
+hostcert="$($PUPPET_BIN/puppet config print cacrl)"
+[[ -e $hostcert ]] || fail "ERROR: primary server cert not found. Run the task crl_truncate to regenerate it"
+
+expiry_date="$($PUPPET_BIN/openssl crl -nextupdate -noout -in "$hostcert")"
+expiry_date="${expiry_date#*=}"
+expiry_seconds="$(date --date="$expiry_date" +"%s")" || fail "Error calculating expiry date from nextupdate"
+
+if (( $(date +"%s") >= expiry_seconds )); then
+  fail "ERROR: The ca crl has expired.  Please pass regen_primary_cert=true to the plan to regenerate it."
```

Comment:
This error message references the `regen_primary_cert` variable.  But, I think we need to change the failure logic here and in the calling plan.

If we want to automatically create a new CRL, then we would need to return the status of the CRL (expired or not expired) in this task.  This would probably need to use the `success` method and return something like:

```
{ "status": "success", "crl_status": "valid" }
```

or a `crl_status` of `expired` if expired.  The `fail` method is more for unexpected results, like if `openssl` returns an error.